### ISSUE TITLE
Dissallow last dragout

### ISSUE THREAD

**Issue by NottocD** [ButchersBoy/Dragablz#154]
In my particular situation I have a main window with several controls around the tab window. I also created a light weight window to receive my torn off tabs. I also can't use FixedCountHeader because when there is more than one tab in the main window I still need to allow the tabs to be reordered.

Default behavior causes my main window frame to disappear if I drag the last tab to a light weight window. So, I wanted to have a behavior similar to the enhancement described here:
https://github.com/ButchersBoy/Dragablz/issues/18

Instead of a simple DissallowDragOut, something more along the lines of DissallowLastDragOut, which is a tab container property.

Investigating the source, I added to TabablzControl this property:

        public static readonly DependencyProperty DissallowLastDragOutProperty = DependencyProperty.Register(
            "DissallowLastDragOut", typeof(bool), typeof(TabablzControl), new PropertyMetadata(default(bool)));

        /// <summary>
        /// Allows a the first adjacent tabs to be fixed (no dragging, and default close button will not show).
        /// </summary>
        public bool DissallowLastDragOut
        {
            get { return (bool)GetValue(DissallowLastDragOutProperty); }
            set { SetValue(DissallowLastDragOutProperty, value); }
        }

And then in the method MonitorReentry right after the FixedHeaderCount test added:
            if (DissallowLastDragOut && sourceTabablzControl.Items.Count == 1)
            {
                return false;
            }

This works for me, hope you like it too.
[TabablzControl.zip](https://github.com/ButchersBoy/Dragablz/files/815200/TabablzControl.zip)

**Comment by ButchersBoy** [comment]
Hi, ``` TabEmptiedResponse ``` is already provided to help you deal with such scenarios:

https://github.com/ButchersBoy/Dragablz/blob/master/Dragablz/TabEmptiedResponse.cs

You can use this by providing a custom ``` IInterTabClient ```

https://github.com/ButchersBoy/Dragablz/blob/master/Dragablz/IInterTabClient.cs

**Comment by ButchersBoy** [comment]
Sorry...that's not going to work is it?

But why not only apply FixedCountHeader to equal 1, _only_ when the tab count is 1?

**Comment by NottocD** [comment]
So I tried that, perhaps I'm not doing it right.

The tabs control is bound to an observable collection. And program startup I added on event for CollectionChanged. And within that, if the Tabs Items list == 1 I set the Tabs FixedHeaderCount = 1, otherwise =0.

When it didn't work I put break points in the ItemDragDelta method and the MonitorReentry method of TabablzControl. What I found is that ItemDragDelta is not called when dragging the last tab. And the check in MonitorReentry says:
if (sourceTabablzControl.Items.Count > 1 && e.DragablzItem.LogicalIndex < sourceTabablzControl.FixedHeaderCount)
When I'm dropping my main window (which now has one tab, onto the simplified window. The sourseTabablzControl has one item in it (failing the > 1 test). So reentry allows the drop to happen. I also checked the FixedHeaderCount field in MonitorReentry, it correctly had a value of 1.
